python/python_helper_sympy.py

```diff
@@ -221,71 +325,206 @@ def evaluate(expr, locals_for_eval={}):
 
     # Convert AST to code and evaluate it with no global expressions and with
     # a whitelist of local expressions
-    locals = {}
-    for key in locals_for_eval:
-        locals = {**locals, **locals_for_eval[key]}
-    return eval(compile(root, '<ast>', 'eval'), {'__builtins__': None}, locals)
+    locals = {
```

Comment:
I think it would be good to separate this out into a `for` loop, similar to the original code.

Reply:
I tend to prefer comprehensions in cases like this, since it compartmentalizes the logic that's used for object creation and makes it clear this dict doesn't need to be modified.  Unfortunately we're not on 3.9, where the dict union syntax makes this much cleaner...